zap/src/main/java/org/zaproxy/zap/extension/history/SiteNotesAddDialog.java

```diff
@@ -0,0 +1,348 @@
+        if (childSiteNodes == null) {
+            childSiteNodes = new JList<>();
+            childSiteNodes.setCellRenderer(new HistoryReferenceRenderer());
+            childSiteNodes.addListSelectionListener(
+                    e -> {
+                        if (!e.getValueIsAdjusting()) {
+                            HistoryReference historyReference = childSiteNodes.getSelectedValue();
+                            if (historyReference != null) {
+                                try {
+                                    setNote(historyReference.getHttpMessage().getNote());
+                                } catch (HttpMalformedHeaderException | DatabaseException ex) {
+                                    LOGGER.error(ex.getMessage(), ex);
+                                }
+                            }
+                        }
+                    });
+        }
+
+        return childSiteNodes;
+    }
+
+    private void updateHrefSelector() {
+        if (siteNode == null) {
+            return;
+        }
+
+        Set<HistoryReference> historyReferencesSet =
+                new TreeSet<>(Comparator.comparingInt(HistoryReference::getHistoryId));
+        fillJSet(historyReferencesSet, siteNode);
+
+        DefaultListModel<HistoryReference> listModel = new DefaultListModel<>();
+        listModel.addAll(historyReferencesSet);
+
+        childSiteNodes.setModel(listModel);
+        if (!historyReferencesSet.isEmpty()) {
+            childSiteNodes.setSelectedIndex(0);
+        }
+    }
+
+    private void fillJSet(Set<HistoryReference> set, SiteNode curSiteNode) {
+        set.add(curSiteNode.getHistoryReference());
+        set.addAll(curSiteNode.getPastHistoryReference());
+
+        for (Object o : Collections.list(curSiteNode.children())) {
+            fillJSet(set, (SiteNode) o);
+        }
+    }
+
+    private ZapTextArea getTxtDisplay() {
+        if (txtDisplay == null) {
+            txtDisplay = new ZapTextArea("", 15, 25);
+        }
+        return txtDisplay;
+    }
+
+    /**
+     * This method initializes btnStart
+     *
+     * @return javax.swing.JButton
+     */
+    private JButton getBtnOk() {
+        if (btnOk == null) {
+            btnOk = new JButton();
+            btnOk.setText(Constant.messages.getString("all.button.save"));
+            btnOk.addActionListener(
+                    e -> {
+                        HistoryReference historyRef = childSiteNodes.getSelectedValue();
+                        if (historyRef != null) {
+                            historyRef.setNote(getTxtDisplay().getText());
+                        }
+                        clearAndDispose();
+                    });
+        }
+        return btnOk;
+    }
+
+    private void clearAndDispose() {
+        setNote("");
+        siteNode = null;
+        dispose();
+    }
+
+    /**
+     * This method initializes btnStop
+     *
+     * @return javax.swing.JButton
+     */
+    private JButton getBtnCancel() {
```

Comment:
It isn't btnStop

Reply:
Better remove, this does not add anything.